### PR TITLE
filter results by DOI name

### LINKED ISSUE #43: /works/<doi> does not return metadata about the DOI I queried
The documentation at https://support.datacite.org/docs/api says I can retrieve *singleton* information about a specific identifier I'd like information about with examples like:

```
curl https://api.datacite.org/works/10.5438/0012
```

When I try it with certain identifiers I actually get a different record than what I requested.

For example:

```
curl https://api.datacite.org/works/10.1098/rsif.2017.0030 # returns information about https://doi.org/10.6084/m9.figshare.c.3706801
curl https://api.datacite.org/works/10.1126/sciadv.1602232 # returns information about https://doi.org/10.6071/m3rp49
curl https://api.datacite.org/works/10.1021/acs.jctc.6b00449 # returns information about https://doi.org/10.7280/d1s887
```

### PR DESCRIPTION
**Pull request by kjgarza**
It filters query results for by DOI, so when there is a request for a DOI you actually get the metadata of that DOI and not the metadata of other DOIS that might be in the metadata. it fixes datacite/spinone#43

### CODE REVIEW

**Review by richardhallett — APPROVED** [review]

File: app/models/work.rb [hunk 1]
```diff
@@ -101,6 +101,7 @@ def dates
   def self.get_query_url(options={})
     if options[:id].present?
       params = { q: options[:id],
+                 fq: "doi:#{options[:id].dump}",
```
Comment (mfenner) [inline app/models/work.rb:104]: What is the `dump` providing?

File: app/models/work.rb [hunk 2]
```diff
@@ -101,6 +101,7 @@ def dates
   def self.get_query_url(options={})
     if options[:id].present?
       params = { q: options[:id],
```
Comment (mfenner) [inline app/models/work.rb:103]: Would this also work without the `q` parameter? There is some duplication in the current code?

File: app/models/work.rb (hunk 1 above)
Comment (kjgarza) [reply app/models/work.rb:104]: It escapes characters. So the DOI name is escaped and then URL encoded ( a few lines down).

File: app/models/work.rb (hunk 2 above)
Comment (kjgarza) [reply app/models/work.rb:103]: Ohh no, it fails with the ones with + sign (if i remove `q`). I would need to figure it why

Comment (mfenner) [reply app/models/work.rb:103]: Ah, ok.

File: app/models/work.rb (hunk 1 above)
Comment (mfenner) [reply app/models/work.rb:104]: What happens when you remove `dump`? As we are not doing this in many similar places in our code.

Comment (mfenner) [reply app/models/work.rb:104]: @kjgarza why is `dump` needed here?

**Pull request merged**